playkit/src/main/java/com/kaltura/playkit/profiler/ExoPlayerProfilingListener.java

```diff
@@ -230,24 +234,10 @@ public void onVideoCodecError(EventTime eventTime, Exception videoCodecError) {
     }
 
     @Override
-    public void onPlayerError(EventTime eventTime, ExoPlaybackException error) {
-        String type = null;
-        switch (error.type) {
-            case ExoPlaybackException.TYPE_SOURCE:
-                type = "SourceError";
-                break;
-            case ExoPlaybackException.TYPE_RENDERER:
-                type = "RendererError";
-                break;
-            case ExoPlaybackException.TYPE_UNEXPECTED:
-                type = "UnexpectedError";
-                break;
-            case ExoPlaybackException.TYPE_REMOTE:
-                type = "remoteComponentError";
-                break;
-        }
-
-        log("PlayerError", field("type", type), "cause={" + error.getCause() + "}");
+    public void onPlayerError(EventTime eventTime, PlaybackException playbackException) {
+        Pair<PKPlayerErrorType, String> exceptionPair = PKPlaybackException.getPlaybackExceptionType(playbackException);
+        String type = exceptionPair.first.toString();
```

Comment:
@GouravSna please add null or empty protection

Reply:
exceptionPair is always non-null in PKPlaybackException class